### PR TITLE
add service deployment mnist

### PR DESCRIPTION
**Pull request by nikp1172**

### CODE REVIEW

File: mnist-classifaction/deploy_model/deploy.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    TruefoundryArtifactSource,
+)
+
+logging.basicConfig(level=logging.INFO)
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--workspace_fqn", type=str, required=True)
+parser.add_argument("--model_version_fqn", type=str, required=True)
+parser.add_argument("--host", type=str, required=True)
+parser.add_argument("--path", type=str, required=False)
```
Comment (debajyoti-truefoundry) [inline mnist-classifaction/deploy_model/deploy.py:19]: Can we add a default path here?

File: mnist-classifaction/deploy_model/deploy.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        memory_limit=500,
+        memory_request=500,
+        ephemeral_storage_limit=600,
+        ephemeral_storage_request=600,
+        cpu_limit=0.3,
+        cpu_request=0.3,
+    ),
+    artifacts_download=ArtifactsDownload(
+        artifacts=[
+            TruefoundryArtifactSource(
```
Comment (debajyoti-truefoundry) [inline mnist-classifaction/deploy_model/deploy.py:41]: I am worried that this example will not work if someone uses Civo or GCP.

File: mnist-classifaction/deploy_model/fastapi_service.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    os.environ.get("MODEL_DOWNLOAD_PATH", "."), "mnist_model.h5"
+)
+model = load_model(model_path)
+
+
+class ImageUrl(BaseModel):
+    url: str = "https://conx.readthedocs.io/en/latest/_images/MNIST_6_0.png"
+
+
+def load_image(img_url):
```
Comment (debajyoti-truefoundry) [inline mnist-classifaction/deploy_model/fastapi_service.py:17]: Add types?

File: mnist-classifaction/deploy_model/gradio_demo.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+from predict import predict_fn, load_model
+import os
+import gradio as gr
+
+model_path = os.path.join(
+    os.environ.get("MODEL_DOWNLOAD_PATH", "."), "mnist_model.h5"
+)
+model = load_model(model_path)
+
+def get_inference(img_arr):
```
Comment (debajyoti-truefoundry) [inline mnist-classifaction/deploy_model/gradio_demo.py:10]: Add types?+

File: mnist-classifaction/deploy_model/deploy.py (hunk 2 above)
Comment (chiragjn) [reply mnist-classifaction/deploy_model/deploy.py:41]: It will still work, right? via init container download?

Comment (nikp1172) [reply mnist-classifaction/deploy_model/deploy.py:41]: yes - I have disabled volume cache. So it will work everywhere

Comment (debajyoti-truefoundry) [reply mnist-classifaction/deploy_model/deploy.py:41]: > I have disabled volume cache.

Right, I missed this.

**Review by debajyoti-truefoundry — APPROVED** [review]

**Review by debajyoti-truefoundry — APPROVED** [review]

**Pull request merged**